sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -543,8 +543,10 @@ private void initializeGatewayConfigurationReader() {
         if (databaseAccount == null) {
             logger.error("Client initialization failed."
                 + " Check if the endpoint is reachable and if your auth token is valid. More info: https://aka.ms/cosmosdb-tsg-service-unavailable-java");
+            logger.error(" More error details: "+this.globalEndpointManager.getLatestDatabaseRefreshError());
```

Comment:
This code is not thread-safe - getLatestDatabaseRefreshError could have been changed (set to null) between these two invocations - instead capture a snapshot - then you can safely log it and throw it if != null - also the refresh error should be passed as inner exception in both cases.